internal/test_cases/run_test_case.go

```diff
@@ -17,8 +18,45 @@ import (
 // It is sent to the "run" command of the executable,
 // the expected outputs are generated using the lox.Run function,
 // With that the output of the executable is matched.
+
 type RunTestCase struct {
 	FileContents string
+	FrontMatter  RunTestCaseFrontMater
+}
+
+type RunTestCaseFrontMater struct {
+	ExpectedErrorType string `yaml:"error_type"`
+}
+
+func NewRunTestCaseFromFileContents(fileContents []byte, filePath string) RunTestCase {
+	contents := string(fileContents)
+
+	if contents[:4] != "---\n" {
+		panic(fmt.Sprintf("CodeCrafters Internal Error: %s has malformed frontmatter: no beginning triple dashes", filePath))
+	}
+
+	contents = contents[4:]
+	endingTripleDashIndex := strings.Index(contents, "\n---\n")
+	if endingTripleDashIndex == -1 {
+		panic(fmt.Sprintf("CodeCrafters Internal Error: %s has malformed frontmatter: no ending triple dashes", filePath))
+	}
+
+	frontMatterBytes := []byte(contents[:endingTripleDashIndex])
```

Comment:
This whole function is a bit confusing because of the bytes / string handling differences. Could we avoid this by using bytes everywhere, and just doing `string(<..>)` in the last `return` block when returning a `RunTestCase`? 

Could use https://pkg.go.dev/bytes#Index.